app/controllers/models_controller.rb

```diff
@@ -6,6 +6,36 @@ class ModelsController < ApplicationController
 
   def index
     @models = resource_base_search_and_page
+    @hosts_counts = Model.hosts_counts
+    @table = @models.map do |m|
+      {
+        :id => m.id,
+        :name => m.name,
+        :vendor_class => m.vendor_class,
+        :hardware_model => m.hardware_model,
+        :hosts_count => @hosts_counts.fetch(m.id, 0),
+        :actions => m.actions_by_user(User.current)
+      }
+    end
+    sort_param = params.fetch(:order, "").split
+    by = (sort_param.length == 2) ? sort_param[0] : nil
+    order = (sort_param.length == 2) ? sort_param[1] : nil
+    @json_models = {
+      :results => @table,
+      :per_page => params.fetch(:per_page, "20"),
+      :page => params.fetch(:page, "1"),
+      :total => @table.length.to_s,
```

Comment:
why not use the API? it seems to have the exact set of data?

Reply:
`hosts_count` and `actions` aren't included in the `API` results and this information is needed when for example we have 6 models, 5 models are shown in each page and we are in page 1. When loading the page, only the first 5 models will have this information, when clicking on sorting by name in descanding, the API will return the last 5 models but the fist model's `actions` and `hosts_count` are missing which ends up with showing the first item as disabled and can't be deleted by the user even though it can be.

I don't know how I can wrap `@table` into `api/v2` response. I know it's available in `Api2::BaseController` or something like that but I don't know if that's possible. 
Anyway, I put it here to see if that format is good enough and improve this more.